File: ros_namespace_example/scripts/namespace_example.py
#! /usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class NamespaceExample(Node):
    def __init__(self):
        super().__init__('namespace_example')
        self.pub_global = self.create_publisher(msg_type=String, topic='/global_pub_topic', qos_profile=1)
        self.pub_private = self.create_publisher(msg_type=String, topic='private_pub_topic', qos_profile=1)
        self.sub_global = self.create_subscription(msg_type=String, topic='/global_sub_topic', qos_profile=1, callback=self.recv_global)
        self.sub_local = self.create_subscription(msg_type=String, topic='local_sub_topic', qos_profile=1, callback=self.recv_local)

    def recv_global(self, msg):
        pass

    def recv_local(self, msg):
        pass


if __name__ == '__main__':
    rclpy.init()
    node_instance = NamespaceExample()
    rclpy.spin(node_instance)

File: ros_namespace_example/src/namespace_example.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

namespace ros_namespace_example {

  class NamespaceExample : public rclcpp::Node {
    public:
      NamespaceExample()
      : rclcpp::Node("namespace_example")
      {
        sub_local_topic_ = create_subscription<std_msgs::msg::String>("local_sub_topic", 1, std::bind(&NamespaceExample::recv_local, this, std::placeholders::_1));
        sub_global_topic_ = create_subscription<std_msgs::msg::String>("/global_sub_topic", 1, std::bind(&NamespaceExample::recv_global, this, std::placeholders::_1));
        pub_local_topic_ = create_publisher<std_msgs::msg::String>("local_pub_topic", 1);
        pub_global_topic_ = create_publisher<std_msgs::msg::String>("/global_pub_topic", 1);
      }

    private:
      rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_global_topic_;
      rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_local_topic_;
      rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_global_topic_;
      rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_local_topic_;

      void recv_local(const std_msgs::msg::String::ConstSharedPtr msg) {

      }

      void recv_global(const std_msgs::msg::String::ConstSharedPtr msg) {

      }
  };
}

int main(int argc, char** argv) {
  rclcpp::init(argc, argv);
  auto node_instance = std::make_shared<ros_namespace_example::NamespaceExample>();
  rclcpp::spin(node_instance);
  rclcpp::shutdown();
  return 0;
}
